src/Server/SSH/SSHBind.cpp

```diff
@@ -0,0 +1,78 @@
+#include "SSHBind.h"
+#include <stdexcept>
+#include <fmt/format.h>
+#include <Common/SSH/clibssh.h>
+
+namespace ssh
+{
+
+SSHBind::SSHBind() : bind_(ssh_bind_new(), &deleter)
+{
+    if (!bind_)
+    {
+        throw std::runtime_error("Failed to create ssh_bind");
+    }
+}
+
+SSHBind::~SSHBind() = default;
+
+SSHBind::SSHBind(SSHBind && other) noexcept : bind_(std::move(other.bind_))
+{
+}
+
+SSHBind & SSHBind::operator=(SSHBind && other) noexcept
+{
+    if (this != &other)
+    {
+        bind_ = std::move(other.bind_);
+    }
+    return *this;
+}
+
+ssh_bind SSHBind::get() const
+{
+    return bind_.get();
+}
+
+void SSHBind::setHostKey(const std::string & key_path)
+{
+    if (ssh_bind_options_set(bind_.get(), SSH_BIND_OPTIONS_HOSTKEY, key_path.c_str()) != SSH_OK)
+        throw std::invalid_argument(fmt::format("Failed setting host key in sshbind due to {}", getError()));
```

Comment:
use `DB::Exception`

Reply:
Changed in `Make ssh objects throw DB::Exception` commit